src/server/replica.cc

```diff
@@ -431,43 +432,40 @@ error_code Replica::InitiateDflySync() {
     shard_flows_[i].reset(new Replica(master_context_, i, &service_));
   }
 
-  SyncBlock sb{num_df_flows_};
+  // Allocate shared, because the error handler might outlive the function scope.
+  auto sync_block = std::make_shared<SyncBlock>();
 
-  AggregateError ec;
+  auto err_handler = [this, sync_block](const auto& ge) {
+    sync_block->Add(num_df_flows_);  // Unblock sync_block.
+    DefaultErrorHandler(ge);         // Close sockets to unblock flows.
+  };
+  if (cntx_.Switch(std::move(err_handler)))
+    return cntx_;
+
+  // Start full sync flows.
   auto partition = Partition(num_df_flows_);
   shard_set->pool()->AwaitFiberOnAll([&](unsigned index, auto*) {
     for (auto id : partition[index]) {
-      if ((ec = shard_flows_[id]->StartFullSyncFlow(&sb)))
-        break;
+      auto ec = shard_flows_[id]->StartFullSyncFlow(sync_block.get(), &cntx_);
+      if (ec)
+        cntx_.Error(ec);
     }
   });
+  RETURN_ON_ERR(cntx_);
 
-  RETURN_ON_ERR(*ec);
-
-  ReqSerializer serializer{sock_.get()};
-
-  // Master waits for this command in order to start sending replication stream.
-  RETURN_ON_ERR(SendCommand(StrCat("DFLY SYNC ", master_context_.dfly_session_id), &serializer));
-
-  base::IoBuf io_buf{128};
-  unsigned consumed = 0;
-  RETURN_ON_ERR(ReadRespReply(&io_buf, &consumed));
-  if (!CheckRespIsSimpleReply("OK")) {
-    LOG(ERROR) << "Sync failed " << ToSV(io_buf.InputBuffer());
-    return make_error_code(errc::bad_message);
+  // Send DFLY SYNC.
+  if (auto ec = SendNextPhaseRequest(); ec) {
+    cntx_.Error(ec);
+    return cntx_;
   }
```

Comment:
I'd really love to do `return cntx_.Error(ec)`, but adding a conversion operator to GenericError makes the comparison operator inside AggregateValue ambiguous, so we'd need to make it a bit more complicated

Not sure if its worth

Reply:
Or we can also do `return cntx_.Error(ec).Code()` without the conversion operator

But this is inconsistent with the context conversion... So it would have to be `return cntx.GetError().Code()` instead of return `cntx` , which is not that bad, because it implicitly states the full return type

Reply:
How do you convert `Context` to error_code ?
I prefer doing such stuff explicitly, i.e. call `GetError` method.